### PR TITLE
feat(assets): added injectAssets HOC & assetProps propType

### PR DESCRIPTION
**Pull request by mhuggins**
## Description
This adds an `injectAssets` HOC and `assetProps` propType.  These can be used to access the `window.NOS.ASSETS` constant, falling back to a default provided by this package.

## Motivation and Context
The `window.NOS.ASSETS` constant was recently introduced in the client.  This brings it up to parity.

## How Has This Been Tested?
Using `create-nos-app` and `yarn link` to use api-functions inside of that app..

## Types of changes
- [ ] Chore (tests, refactors, and fixes)
- [x] New feature (adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** guidelines and confirm that my code follows the code style of this project.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Closing issues
N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,48 @@ yarn add @nosplatform/api-functions
...
+    const { nos, assets } = this.props;
+    const balance = await nos.getBalance({ asset: assets.NEO });
+    console.log('NEO Balance:', balance);
+  }
+};
+
+export default compose(
+  injectNOS,
+  injectAssets
+)(ShowBalance);
```
Comment (DalderupMaurice) [inline README.md]: This looks so beautiful! 
We'll update this in the create-nos-dapp starter kits aswell later

File: es6/executor.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
 /** @const {boolean} exists check whether nOS is defined */
-export const exists = !!window.NOS && !!window.NOS.V1;
+const exists = !!window.NOS && !!window.NOS.V1;
```
Comment (DalderupMaurice) [inline es6/executor.js]: We actually did export this for users to have a mechanic to check if the nOS API is injected or not. I think many people are depending on this :P

File: package.json [hunk 3]
```diff
@@ -30,15 +31,15 @@
   },
   "homepage": "https://github.com/nos/api-functions#readme",
   "dependencies": {
-    "prop-types": "15.6.1"
+    "prop-types": "15.6.2"
   },
   "devDependencies": {
     "@babel/cli": "^7.0.0-beta.46",
```
Comment (DalderupMaurice) [inline package.json]: Could you update babel aswell to 7 :P There should be no changes in usage

File: package.json [hunk 4]
```diff
@@ -30,15 +31,15 @@
...
     "@babel/core": "^7.0.0-beta.46",
     "@babel/preset-env": "^7.0.0-beta.46",
     "@babel/preset-react": "^7.0.0-beta.47",
-    "react": ">=16.3.0",
-    "prettier": "^1.12.1"
+    "prettier": "^1.12.1",
```
Comment (DalderupMaurice) [inline package.json]: What's your experience with prettier? :P 
We're not using it in the client currently, but we've had this discussion at some point

**Comment by DalderupMaurice** [comment]
Could we rename `es6` to something more generic, although it's not related to this PR xD

File: es6/executor.js (hunk 2 above)
Comment (jeroenptrs) [reply es6/executor.js]: Yeah `exists` was exported into the basic nOS namespace and now it's gone. Might be a nifty built-in trick that we should keep.

File: es6/react/props.js [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
+  getBalance: func.isRequired,
+  claimGas: func.isRequired,
+  send: func.isRequired,
+  testInvoke: func.isRequired,
+  invoke: func.isRequired,
+  getStorage: func.isRequired
 });
 
-export default nosProps;
+export const assetProps = objectOf(string);
```
Comment (jeroenptrs) [inline es6/react/props.js]: Would like to see an isRequired here 👍

File: es6/assets.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+/** @const {boolean} exists check whether assets constant is defined */
+const exists = !!window.NOS && !!window.NOS.ASSETS;
```
Comment (jeroenptrs) [inline es6/assets.js]: Would love to see this exported as `assetsExist` or something!

File: package.json (hunk 3 above)
Comment (jeroenptrs) [reply package.json]: I'd hold off updating babel and do that in a seperate PR. Beta to RC to final might still involve things that break and I want to be sure 💪

File: package.json (hunk 4 above)
Comment (jeroenptrs) [reply package.json]: +1 we really should

File: es6/assets.js (hunk 6 above)
Comment (DalderupMaurice) [reply es6/assets.js]: Doesnt seem so usefull in my eyes - it's also initialized if it doesn't exist, so it always exists

Comment (jeroenptrs) [reply es6/assets.js]: True.

**Comment by mhuggins** [comment]
@jeroenptrs @DalderupMaurice I updated to restore the `exists` boolean on the `nos` object.  I think that was really the only blocking comment.  Let me know what you think.  This PR is a blocker for adding the new `getLastBlock` function, so it'd be good to get this in if we can.

File: es6/react/props.js (hunk 5 above)
Comment (mhuggins) [reply es6/react/props.js]: Putting an isRequired here would be like the equivalent of putting it onto [line 11](https://github.com/nos/api-functions/pull/7/files/072aa9af2d95ca2abb519b8c0ed9122dbab8217b#diff-0189fe5a3008eda6c36c0d9fdeb8881dR11).  We're just defining the shape, it should be up to the implementor whether they want to require it when using it or not.

**Pull request merged**